File: frontend/src/types/recommendation.ts
export interface ObjectProfile {
  object_name: string;
  display_name?: string;
  material?: string;
  condition?: string;
  confidence?: number;
}

export interface BudgetConstraint {
  min: number;
  max: number;
  currency?: string;
}

export interface RecommendationRequest {
  object_name: string;
  object?: ObjectProfile;
  goal: string;
  custom_goal?: string;
  tools: string[];
  materials: string[];
  budget_min: number;
  budget_max: number;
  budget?: BudgetConstraint;
  difficulty: string;
  max_time_minutes: number;
  time_minutes?: number;
}

export interface EstimatedCost {
  min: number;
  max: number;
  currency?: string;
}

export interface RecommendationItem {
  project_id: string;
  name: string;
  title?: string;
  description: string;
  match_score: number;
  why_it_matches?: string[];
  difficulty: string;
  estimated_time_minutes: number;
  estimated_cost_min: number;
  estimated_cost_max: number;
  estimated_cost?: EstimatedCost;
  matched_factors: string[];
  missing_requirements: string[];
  tools_needed?: string[];
  materials_needed?: string[];
  steps?: string[];
  is_top_match?: boolean;
  image_color?: string;
}

export interface RecommendationResponse {
  success: boolean;
  object_name: string;
  top_recommendation: RecommendationItem | null;
  recommendations: RecommendationItem[];
  message?: string;
}

export interface ProjectDetails {
  success: boolean;
  project_id: string;
  name: string;
  description: string;
  supported_objects: string[];
  goals: string[];
  required_tools: string[];
  optional_tools: string[];
  required_materials: string[];
  optional_materials: string[];
  difficulty: string;
  estimated_time_minutes: number;
  estimated_cost_min: number;
  estimated_cost_max: number;
  safety_notes: string[];
  steps: string[];
}

File: backend/app/schemas/recommendation.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel

class ObjectProfile(BaseModel):
    object_name: str
    display_name: Optional[str] = None
    material: Optional[str] = "unknown"
    condition: Optional[str] = "used"
    confidence: Optional[float] = 0.95

class BudgetConstraint(BaseModel):
    min: float = 0
    max: float = 50
    currency: str = "INR"

class RecommendationRequest(BaseModel):
    object_name: str
    object: Optional[ObjectProfile] = None
    goal: str = "gardening"
    custom_goal: Optional[str] = None
    tools: List[str] = []
    materials: List[str] = []
    budget_min: float = 0
    budget_max: float = 50
    budget: Optional[BudgetConstraint] = None
    difficulty: str = "easy"
    max_time_minutes: int = 30
    time_minutes: Optional[int] = None

class EstimatedCost(BaseModel):
    min: float = 0
    max: float = 50
    currency: str = "INR"

class RecommendationItem(BaseModel):
    project_id: str
    name: str
    title: Optional[str] = None
    description: str
    match_score: int          # 0 - 100
    why_it_matches: List[str] = []
    difficulty: str
    estimated_time_minutes: int
    estimated_cost_min: float = 0
    estimated_cost_max: float = 50
    estimated_cost: Optional[EstimatedCost] = None
    matched_factors: List[str] = []
    missing_requirements: List[str] = []
    tools_needed: List[str] = []
    materials_needed: List[str] = []
    steps: List[str] = []
    is_top_match: bool = False
    image_color: str = "bg-emerald-100"

class RecommendationResponse(BaseModel):
    success: bool
    object_name: str
    top_recommendation: Optional[RecommendationItem] = None
    recommendations: List[RecommendationItem] = []
    message: Optional[str] = None

class ProjectDetailsResponse(BaseModel):
    success: bool
    project_id: str
    name: str
    description: str
    supported_objects: List[str] = []
    goals: List[str] = []
    required_tools: List[str] = []
    optional_tools: List[str] = []
    required_materials: List[str] = []
    optional_materials: List[str] = []
    difficulty: str
    estimated_time_minutes: int
    estimated_cost_min: float = 0
    estimated_cost_max: float = 50
    safety_notes: List[str] = []
    steps: List[str] = []
